### PR TITLE
CHECKOUT-1967 Resolve Accessibility Issues

### PR DESCRIPTION
**Pull request by icatalina**
#### What?

Adding some missing customisation options for Optimized Checkout so merchants can fully tailor the shopper experience and be fully accessible.

#### Tickets / Documentation

- [CHECKOUT-1967](https://jira.bigcommerce.com/browse/CHECKOUT-1967)

#### Screenshots (if appropriate)

On their way 😊

### CODE REVIEW

File: assets/scss/optimized-checkout.scss [hunk 1]
```diff
@@ -171,15 +224,74 @@ a {
...
+}
 //
 // Form Input
-// Background & Border Color
+// Background, Border, Shadow and Text Color
 // Styles form fields.
 // -----------------------------------------------------------------------------
 
-.optimizedCheckout-form-input {
+$optmizedCheckout-inputBoxShadow: inset 0 1px 1px stencilColor("optimizedCheckout-formField-shadowColor");
```
Comment (davidchin) [inline assets/scss/optimized-checkout.scss]: You have a spelling mistake here: `optmizedCheckout` -> `optimizedCheckout` :)

Comment (davidchin) [inline assets/scss/optimized-checkout.scss]: Also, I think we have a specific way of naming SASS variables (i.e.: https://github.com/bigcommerce/sass-style-guide#variables). Do you want to follow that?

**Comment by mcampa** [comment]
The schema file is crazy big right now in Cornertone, and ThemeEditor was not really designed to support so many fields from the schema. This is going to slow down ThemeEditor. 

I'm not opposing this change, just something to think about. If you have the chance, please take a look a TE to see how we can improve performance.

**Comment by davidchin** [comment]
@mcampa very good point. I agree we can look into improving the performance of TE separately. :)

**Comment by icatalina** [comment]
I tackled those issues... please have a look 🙂

**Comment by junedkazi** [comment]
can we get some screenshots & a QE sign off since it has lots of changes to the schema.

**Comment by davidchin** [comment]
Hey @junedkazi, thanks for checking. We already reached out to Jully, hoping that one of the QEs / designers from Stencil team can help verify the changes in the upcoming sprint. There's a separate ticket tracking the QA work (STENCIL-3751). :)

**Comment by mjschock** [comment]
@icatalina - could you rebase into one commit and add a changelog entry?

**Comment by icatalina** [comment]
@mjschock For sure...

**Comment by icatalina** [comment]
I let it like this because it is easier to review...

**Comment by icatalina** [comment]
Hi @mjschock!,

All squashed and change log added. I've kept a copy not squashed just in case it makes it easier to review: https://github.com/icatalina/cornerstone/tree/CHECKOUT-1967-unsquash

Could you please have a look and confirm you're happy with these changes?

Kind Regards,
Ignacio Catalina

**Comment by icatalina** [comment]
Some screenshots:

|Light|Before|After|
|----|----|----|
|Home|light home org|light home new|
|Product|light product org|light product new|
|Cart|light cart org|light cart new|
|Checkout|light checkout org|light checkout new|

|Warm|Before|After|
|----|----|----|
|Home|warm home org|warm home new|
|Product|warm product org|warm product new|
|Cart|warm cart org|warm cart new|
|Checkout|warm checkout org|warm checkout new|

|Dark|Before|After|
|----|----|----|
|Home|dark home org|dark home new|
|Product|dark product org|dark product new|
|Cart|dark cart org|dark cart new|
|Checkout|dark checkout org|dark checkout new|

File: schema.json [hunk 2]
```diff
@@ -1,2498 +1,2626 @@
-[
```
Comment (mjschock) [inline schema.json]: @icatalina - is there a way to get the changes in here without causing the entire schema to differ?

**Comment by icatalina** [comment]
I tackled @mjschock concerns... I'll raise a different PR for changing the `schema.json` format to `unix`.

**Comment by davidchin** [comment]
I think it makes to squash commits that cannot be checked out individually (i.e.: WIP commits). If each commit is an incremental change that is descriptive and comprehensible, I don't think it's strictly necessary to squash them into one.
* Firstly, it's easier for reviewers to review the changes.
* Secondly, it's easier for the team to look at the history and go back in time when things break.
* Thirdly, it's easier to break the PR into multiple ones if it contains multiple unrelated changes which should be submitted separately.

Having said that, sometimes, it's actually easier to review the changes or check the history if the commits are squashed into one (or fewer ones). In that case, yes, we should squash the commits. So I think it really depends on the situation. We shouldn't squash the commits just because we need to follow a rule. Anyway, just my two cents 😄

**Pull request merged**